FreeRTOS/FreeRTOS-Kernel
# Add macro taskTASK_IS_RUNNING_OR_SCHEDULED_TO_YIELD

Description
-----------
* Add taskTASK_IS_RUNNING_OR_SCHEDULED_TO_YIELD macro to align single core and SMP
* Update for explicit precedence in vTaskDelete
* Update the comment to delete running or yielding task to align SMP implementation

Test Steps
-----------
N/A

Checklist:
----------



Related Issue
-----------
Address #750  comment in line [1922](https://github.com/RichardBarry/FreeRTOS-Kernel/blob/814f078563eaf20ba9f59bd36cf48d85ae5e3185/tasks.c#L1922)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
